packages/notebook/src/notebookfooter.ts

```diff
@@ -32,11 +32,13 @@ export class NotebookFooter extends Widget {
       case 'click':
         this.onClick();
         break;
-      case 'keydown':
-        if ((event as KeyboardEvent).key === 'ArrowUp') {
+      case 'keydown': {
+        const key = (event as KeyboardEvent).key;
+        if (key === 'ArrowUp' || key === 'k') {
           this.onArrowUp();
           break;
         }
+      }
```

Comment:
Thank you for highlighting it.

Ideally we would not have any of the keys hard-coded here and instead just check against keys defined for `notebook:move-cursor-up`, as users may want to remap their shortcuts (and in some cases when the notebook component is reused the <kbd>k</kbd> may not event be defined as cycling down). You can get the keys with:

```ts
const binding = app.commands.keyBindings.find(
    b => b.command === 'notebook:move-cursor-up'
);
binding.keys
```

The shortcuts themselves are defined here:

https://github.com/jupyterlab/jupyterlab/blob/c4de22b707eecc17c5d5c181c7b9991242abf9db/packages/notebook-extension/schema/tracker.json#L510-L518

Reply:
Thanks for the pointer. I think we may need to do some additional work due to casing (i.e. I'm guessing that on the command it will be registered as `'K'` while on the event itself it will look like `'k'`). If so, would it be OK to check against the lowercase variants of everything?

Reply:
Yes, I think so.

Reply:
Hmm just took another look at this today, seems like we don't have easy access to `app` / `JupyterFrontEnd` object from here. We would have to thread it down through `Notebook` to `NotebookFooter`, unless there's some magic bypass attached to `window`. Doing so seems like it would pollute a lot of public interfaces. Maybe we can merge the hack to at least get it working for the default shortcuts, and think about how to do a deeper design / integration with the keybinding system as a followup?

(Feel free to correct anything I misunderstood, my knowledge of the JupyterLab codebase is fairly superficial and based on very hacky extension development :) )

Reply:
Right. Maybe we should do this the other way round and teach `notebook:move-cursor-up` to handle the footer? It invokes `NotebookActions.selectAbove` defined here:

https://github.com/jupyterlab/jupyterlab/blob/42fb46470fb747ec1a8e6ad9a32ed11a0d4991a7/packages/notebook/src/actions.tsx#L846-L852

We could expose `footer` on `Notebook` and if it is focused, handle blurring it in `selectAbove()`?

Reply:
Trying this now; hmm is there some upstream build issue? Can't install packages locally, and seems like CI may have the same problem.

Reply:
On CI only `check_release` job is expected to fail due to transitive GitHub API failure, all others jobs are passing elsewhere. Logs show a ts error in build:

```ts
  ../notebook/src/notebookfooter.ts(38,16): error TS2339: Property 'onArrowUp' does not exist on type 'NotebookFooter'.
```

It is still used in the fragment we are commenting on, so just removing it should fix it:

```suggestion
```

Reply:
Oh yikes, thanks for catching that!

Reply:
Finally managed to build jupyterlab locally in order to test e2e, and seems that `notebook:move-cursor-up` is not triggered when the footer is in focus, so unfortunately that approach doesn't work either. I did test out the original hacky approach of just checking for 'k' directly in the footer, and it seemed to work. I think to do this "properly" we would either need to figure out how to make the footer aware of the command registry, as in the original suggestion, or we would need to figure out if there's a way to make sure the notebook actions are triggered when the footer is in focus.

Reply:
So the selector in schema (the tracker.js file linked above) needs to be modified. There is another PR which will change these selectors from Notebook:focus to something more reasonable. Let's wait here for that PR, I will let you know once it's merged.

Reply:
Checking back in here -- I took a look at trunk this morning at it seems the selectors are still the same. I'm guessing that the changes to fix that are fairly involved and will take some time to merge. Wanted to get your thoughts on hardcoding the shortcuts for now to take focus off the footer when the user presses `j` and `k` so that we can at least get the usability improvement in for the common case so that users (and selfishly, me in particular :) ) can start to enjoy it.

Reply:
That's fair. Could we special case the selector for `notebook:move-cursor-up` so that it matches either `.jp-Notebook:focus` or `.jp-Notebook-footer`? This is maybe `.jp-Notebook:focus,.jp-Notebook-footer`. I am not 100% sure if this will work, but seems worth trying.

Reply:
So I wasn't able to use a selector with commas because it looks like it's a lumino selector and not a proper css selector (and lumino disallows commas), but what did work was to add an entirely new entry in the array for `.jp-Notebook-footer`. I did also have to check in the handler if the footer got unblurred, because otherwise it would both unfocus the footer and move up to the second-to-last cell (rather than staying at the last cell, which is what I expect), but overall it seems to be working. Thanks for the suggestion!

The only downside is that, in the keyboard shortcuts editor, there are now two entries for `Select Cell Above`, and both need to be edited if the user wants a working custom shortcut 😅 

Maybe @jasongrout knows how to get "boolean or" behavior in a lumino selector?

Reply:
I see, I guess the restriction in lumino could be lifted but this is a larger refactor due to other code that would need to be updated (e.g. specificity calculation). A solution is to add a class or data attribute which indicates the target; something like `[data-jp-traversable]:focus` and then the notebook gets `data-jp-traversable` attribute.

Reply:
I will push a commit in a minute.

Reply:
See https://github.com/jupyterlab/jupyterlab/pull/14796/commits/21659f7ef5b3cc0481a8a629a7aaf1dc8c2a0cbc.